### ISSUE TITLE
tilt does not understand kubernetes service generated by helm

### ISSUE THREAD

**Issue by mateka** [tilt-dev/tilt#3732]
My setup:
1. I am building local rust application and want to deploy it like in c# example, but
2. I have done `helm create app` and docker image name is under `spec.template.spec.containers.image`. 

If I change something in rust application source code, tilt rebuilds app but not the docker image and does not deploy app on kubernetes. I have found that, [docker_build](https://docs.tilt.dev/api.html#api.docker_build) expects image name to be in `spec.container.image`, which does not corresponds to helm default deployment configuration. Is it possible for tilt to understand kubernetes deployments generated by helm?

**Comment by landism** [comment]
> Is it possible for tilt to understand kubernetes deployments generated by helm?

Yep! There are plenty of people successfully using Tilt with helm charts.

Can you share your Tiltfile and/or some logs?

Just to confirm, if Tilt is failing to find the image name in the yaml, there should be a warning like:
```
WARNING: Image not used in any Kubernetes config:
    ✕ my-rust-image
```

Are you seeing this?

If not, the other main possibility is that the `local_resource` that generates your rust binary is not writing to a path that is watched for your `docker_build` (if you've configured your Tiltfile like step 2 or 3 of the c# example, which it sounds like you have). For each `docker_build`, Tilt watches the path specified as its build context (the second arg)

>  I have found that, docker_build expects image name to be in spec.container.image

May I ask what led to this conclusion? I'd like to improve the clarity of the docs around this. Tilt will find image names in any .container.image anywhere in the yaml. The C# example you mention [specifies its image](https://github.com/tilt-dev/tilt-example-csharp/blob/master/3-live-update/kubernetes.yaml#L17) under `spec.template.spec.containers.image`, which matches the path your helm template is using.

**Comment by mateka** [comment]
Thanks for a fast response! I have found that in https://docs.tilt.dev/api.html#api.docker_build ref section:

> If this image will be used in a k8s resource(s), this ref must match the spec.container.image param for that resource(s).

No, I'm not seeing warning about image. I have done one more thing: I have changed build directory from 'auto-target' into 'out' and then everything magically works. When I try to use 'auto-target' again, tilt builds binary but not the docker image. Maybe Windows or building by wsl break something?

**Comment by landism** [comment]
> I have found that in https://docs.tilt.dev/api.html#api.docker_build ref section

Thanks! We'll clear that up.

> When I try to use 'auto-target' again, tilt builds binary but not the docker image.

Would you mind sharing some logs or Tiltfile? And/or could you talk a bit about how you've configured your resources and what you expect to happen? (e.g., file A changes which triggers local_resource B which outputs file C which triggers docker_build D)

**Comment by mateka** [comment]
Yes, I can. I was playing with tilt and rust after last KubeCon and wanted to create rust&tilt hello world example. I will publish it on GitHub, after I clean up some dead code.
